addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -239,17 +239,20 @@ var SnippetEditor = Widget.extend({
      * Removes the associated snippet from the DOM and destroys the associated
      * editor (itself).
      */
-    removeSnippet: function () {
+    removeSnippet: async function () {
         this.toggleOverlay(false);
         this.toggleOptions(false);
 
-        this.trigger_up('call_for_each_child_snippet', {
-            $snippet: this.$target,
-            callback: function (editor, $snippet) {
-                for (var i in editor.styles) {
-                    editor.styles[i].onRemove();
-                }
-            },
+        await new Promise (resolve => {
+                this.trigger_up('call_for_each_child_snippet', {
+                $snippet: this.$target,
+                callback: async (editor, $snippet) => {
+                    for (var i in editor.styles) {
+                        editor.styles[i].onRemove();
```

Comment:
Currently, we don't await onRemove or onClone, meaning they can't be async (unlike most other option methods), should I await both of those to allow that?

Reply:
@sdegueldre I'd like to make all the onXXX method async but we'll do that in a separate PR in master. You can rebase this PR though since making removeSnippet async was made by felix' work, merged this week-end.

Reply:
Felix's PR targets 12.3 and you disabled forward-porting so the changes won't be in 13.2, this work was done with 13.2 up to date.

Reply:
> I'll not forward-port this through the bot but instead myself with the async await form.

Merged https://github.com/odoo/odoo/pull/44900, just not forward-ported yet because of conflict

Reply:
Alright, I have subscribed to the 13.2 forward port PR, will rebase and ping you as soon as it's merged.